x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/profile/RestSuggestProfilesAction.java

```diff
@@ -31,12 +32,22 @@ public class RestSuggestProfilesAction extends SecurityBaseRestHandler {
 
     static final ConstructingObjectParser<Payload, Void> PARSER = new ConstructingObjectParser<>(
         "suggest_profile_request_payload",
-        a -> new Payload((String) a[0], (Integer) a[1])
+        a -> new Payload((String) a[0], (Integer) a[1], (PayloadHint) a[2], (String) a[3])
+    );
+
+    @SuppressWarnings("unchecked")
+    static final ConstructingObjectParser<PayloadHint, Void> HINT_PARSER = new ConstructingObjectParser<>(
+        "suggest_profile_request_payload_hint",
+        a -> new PayloadHint((List<String>) a[0], (Map<String, Object>) a[1])
```

Comment:
Yes using `Object` is due to having a mix of `List<String>` and `String`. Otherwise we can parse for more specific Maps, e.g. `Map<String, List<String>>`. But the error message will be different, parsing error vs validation error. Often times, validation error is more specific and easier to understand as oppose to parsing error which just tells you something like "expecting a start of object, but get xxx" etc.